packages/@aws-cdk/aws-logs/lib/metric-filter.ts

```diff
@@ -43,6 +43,7 @@ export class MetricFilter extends Resource {
     new CfnMetricFilter(this, 'Resource', {
       logGroupName: props.logGroup.logGroupName,
       filterPattern: props.filterPattern.logPatternString,
+      filterName: props.filterName ?? undefined,
```

Comment:
i think we can just do `filterName: props.filterName` here. The only difference i can think of is how this handles `props.filterName == ''`, but I don't see why that should match to an `undefined` filterName.